File: Data Structures & Algorithms/search-2d-matrix/submission-1.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int n = matrix[0].size();
        int m = matrix.size();

        int l = 0;
        int r = n * m - 1;

        while(l <= r) {
            int tmp = l  + (r - l) / 2;
            cout << tmp << endl;
            int i = tmp / n;
            int j = tmp % n;

            if(matrix[i][j] == target) return true;
            else if(matrix[i][j] > target) r = tmp - 1;
            else l = tmp + 1;
        }

        return false;
    }
};

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        while(left <= right):
            tmp = (left + right) // 2
            print(f"{left}-{tmp}-{right}")
            if nums[tmp] == target:
                return tmp
            elif target > nums[tmp]:
                left = tmp + 1
            else:
                right = tmp - 1
        
        return -1